File: PHP/70. Climbing Stairs.php
<?php
class Solution {

    /**
     * @param Integer $n
     * @return Integer
     */
    function climbStairs($n) {
        $t = array(1, 1);
        for($i = 2; $i <= $n; $i++){
            $t[$i] = $t[$i - 1] + $t[$i - 2];
        }
        return $t[$n];
    }
}

File: C++/70. Climbing Stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        long long a = 1, b = 2, t;

        for (int i = 1; i < n; i++) {
            t = b;
            b += a;
            a = t;
        }

        return a;
    }
};
